### ISSUE TITLE
Change river shape when infrastructure is built

### ISSUE THREAD

**Issue by hugomflavio** [longturn/freeciv21#786]
As it is, the client selects the correct river sprite depending on the river/lake/sea connections of the neighbouring tiles.

I would like it to be able to account for the presence of cities, ruins, airstrips and fortresses so that, when any of these structures is present, the river shape changes. 

Simple draft provided:
![image](https://user-images.githubusercontent.com/40389021/147391360-8a92ac66-3e62-4fd3-b265-0fcc146d6735.png)

This would mean that the number of river sprites doubles. One sprite for the 'simple' river and one sprite for the 'moat' river.

Finally, the city walls should adapt to the presence of the river too, so we would need to be able to select between two different wall sprites.

**Comment by lmoureaux** [comment]
One thing that always bugged me with rivers is the appearance of mountain+river tiles:
![image](https://user-images.githubusercontent.com/22327575/147391731-f45bb5cc-fbb2-4712-bb63-db98f508f813.png)

**Comment by hugomflavio** [comment]
Agreed! It should also be possible to make the river sprite tile-dependent, even if there is no infrastructure in it at all. I would also add that it doesn't make much sense for rivers to go grassland - mountain - forest. If anything, the mountain tile should have two springs (one in each side). This should have consequences for river-moving units too (e.g make mountains impassable by triremes?).

**Comment by lmoureaux** [comment]
I used the map editor to make this screenshot :D
